gauth.go

```diff
@@ -42,22 +47,45 @@ func main() {
 
 	currentTS, progress := gauth.IndexNow()
 
-	tw := tabwriter.NewWriter(os.Stdout, 0, 8, 1, ' ', 0)
-	fmt.Fprintln(tw, "\tprev\tcurr\tnext")
-	for _, record := range cfg {
-		name, secret := record[0], record[1]
-		prev, curr, next, err := gauth.Codes(secret, currentTS)
-		if err != nil {
-			log.Fatalf("Generating codes: %v", err)
+	if *useCSV {
+		cw := csv.NewWriter(os.Stdout)
+		for _, record := range cfg {
+			name, secret := record[0], record[1]
+			_, curr, next, err := gauth.Codes(secret, currentTS)
+			if err != nil {
+				log.Fatalf("Generating codes: %v", err)
+			}
+			if err := cw.Write([]string{name, curr, next, strconv.Itoa(30 - progress)}); err != nil {
+				log.Fatalf("Printing CSV: %v", err)
```

Comment:
The impulse to check errors is commendable. That said: I don't think there is much practical benefit to checking the error results from these calls.

The circumstances under which writes to a pipe can fail are highly unlikely, and if the process can't write to its standard descriptors for some reason, there is little reason to expect the log output will survive either. On that basis, I recommend you not bother with the noise of these checks.

Did something specific inspire you to add these?

Reply:
TL;DR (would probably be brokenProcessor starting gauth and checking its exit code):

    gauth 2> errors | brokenProcessor

brokenProcessor could be close its input prematurely for example, which could lead to annoying bugs hard to track if gauth swallows failures.

Reply:
Most Unix tools do not log anything for writing on a broken pipe—that's actually working as intended. For example, if we pipe the output of a tool into `head -5`, it would be annoying if the tool complained because `head` stopped reading early. And if the consumer _fails_ in some way, arguably that consumer should log an error, rather than expecting the upstream to do so.

That said: It's not _wrong_ to check these, just unusual.

Reply:
> That said: It's not wrong to check these, just unusual.

Not really, no. cat from GNU:
https://github.com/coreutils/coreutils/blob/master/src/cat.c#L187

xxd from vim:
https://github.com/vim/vim/blob/master/src/xxd/xxd.c#L376